src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+### fluid.transform(source, [fn1, fn2, ...])
+
+Return a list or hash of objects, transformed by one or more functions. Similar to
+[`jQuery.map`](http://api.jquery.com/jquery.map/), but accepts an arbitrary list of transformation functions and does not explode on null sources. 
+`fluid.transform` on a non-array and non-object (e.g. `null`) is a no-op.
+
+* `source {Array|Object}` The initial container of objects to be transformed. If the source is neither an array nor an object, it will be returned untransformed
+* `fn_i {Function} (element {Any}, index {Number|String})→Any` (arbitrary number accepted) An arbitrary number of optional further arguments,
+all of type `Function`, accepting the signature `(element, index)`, where `element` is the
+list member to be transformed, and `index` is its index. Each function will be
+applied in turn to each member, which will be replaced by the return value from the function.
+* Returns: `{Array|Object}` The finally transformed structure, where each member has been replaced by the original member acted on by the function or functions.
+
+### fluid.accumulate(list, fn, initial)
+
+Scan through a list of objects, "accumulating" a value over them
+(may be a straightforward "sum" or some other chained computation). "accumulate" is the name derived
+from the C++ STL, other names for this algorithm are "reduce" or "[fold](https://en.wikipedia.org/wiki/Fold_%28higher-order_function%29)".
+
+* `list {Array}` The list of objects to be accumulated over.
+* `fn {Function: (object {Any}, total {Any-Total}, index {Integer}) → Any-Total}` An "accumulation function" accepting the signature `(object, total, index)` where
+`object` is the list member, `total` is the "running total" object (which is the return value from the previous invocation of `fn` or else `initial`)
+, and index is the index number.
+* `arg {Any-Total}` The initial value for the "running total" object.
+* Returns: `{Any-Total}` The final running total object as returned from the final invocation of the function on the last list member.
+
+
+### fluid.remove_if(source, fn[, target])
+
+Scan through a list or hash of objects, removing those which match a predicate. The source structure *** will be modified in place ***.
+
+* `source {Array|Object}` The list or hash of objects to be scanned over.
+* `fn {Function: (object {Any}, index {Number|String}) → Booleanish }` A predicate function determining whether an element should be
+removed. This accepts the standard signature `(object, index)`. If the predicate returns a "truthy" value, the corresponding object
+will be removed from the source structure.
+* `target {Array|Object}` [optional] A target object of the same type as `source`, which will
+receive any objects removed from it.
+* Returns: `{Array|Object}` Either `target`, containing the removed elements, if it was supplied, or else `source`
+modified by the operation of removing the matched elements.
+
+
+### fluid.getMembers(holder, path)
```

Comment:
The explanation for this function isn't quite clear. I'm not sure how to improve it, since it's still not clear to me what it does :-) "each top-level member": does that mean that the path is relative to the top level? what would the paths be in the resulting object? what does 'path' do if the holder is an array?...